packages/components/src/Actions/ActionButton/ActionButton.component.js

```diff
@@ -212,14 +208,11 @@ ActionButton.propTypes = {
 	model: PropTypes.object, // eslint-disable-line react/forbid-prop-types
 	name: PropTypes.string,
 	onClick: PropTypes.func,
-	overlayId: PropTypes.string,
-	overlayComponent: PropTypes.element,
-	overlayPlacement: OverlayTrigger.propTypes.placement,
-	overlayRef: PropTypes.func,
-	tooltipPlacement: OverlayTrigger.propTypes.placement,
+	tooltipPlacement: BaseOverlayTrigger.propTypes.placement,
 	t: PropTypes.func,
 	tooltip: PropTypes.bool,
 	tooltipLabel: PropTypes.string,
+	...overlayPropTypes,
```

Comment:
@jmfrancois Could you explain me some time why flat component props is a good thing ? I can't find any good reason other than a false simplicity reason because of not nesting.

In the other hand this generate a potential props override if conflict of name :
   - So need to check carefully mentally that's not the case across multiple components/files (cerebral useless pain)
   - Or need to name things super specifically, which break the encapsulation scope by including parent name in children things (component name in props prefix) => which is actually more or less the case in ` packages/components/src/OverlayTrigger/OverlayTrigger.component.js`. Plus add more verbosity and useless informations to mentally process/truncate

And it also in my opinion turn the props interface to a more complicated approach because when we use it we need to check all props one at a time. Rather than in the other case, we can check if we need a feature and discard all informations of this context quickly if no needed (example here: tooltip, overlay, are two specific feature of the action button, it's not directly related to). And if we need a feature, **only** at this time, we can follow the rabbit hole.